numpy/lib/function_base.py

```diff
@@ -327,6 +327,39 @@ def _hist_bin_scott(x):
     return (24.0 * np.pi**0.5 / x.size)**(1.0 / 3.0) * np.std(x)
 
 
+def _hist_bin_ise(x):
+    """
+    Histogram bin estimator based on minimizing the estimated integrated squared error (ISE).
+
+    The number of bins is chosen by minimizing the estimated ISE against the unknown true distribution.
+    The ISE is estimated using cross-validation and can be regarded as a generalization of Scott's rule.
+    https://en.wikipedia.org/wiki/Histogram#Scott.27s_normal_reference_rule
+
+    Parameters
+    ----------
+    x : array_like
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+
+    n = x.size
+    ptp_x = np.ptp(x)
+    if n <= 1 or ptp_x == 0:
+        return ptp_x
+
+    def jhat(nbins):
+        hh = ptp_x / nbins
+        N_k = np.histogram(x, bins=nbins)[0] / n
+        return (2 - (n + 1) * N_k.dot(N_k)) / (n - 1) / hh
```

Comment:
Ah, very good. I would suggest using `p_k` then (which matches Wasserman's notation, maybe I should update the Wiki page) as the variable name instead of `N_k`.